awx/ui_next/src/components/JobCancelButton/JobCancelButton.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useCallback, useState } from 'react';
+import { t } from '@lingui/macro';
+import { MinusCircleIcon } from '@patternfly/react-icons';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { getJobModel } from '../../util/jobs';
+import useRequest, { useDismissableError } from '../../util/useRequest';
+import AlertModal from '../AlertModal';
+import ErrorDetail from '../ErrorDetail';
+
+function JobCancelButton({
+  job = {},
+  errorTitle,
+  title,
+  showIconButton,
+  errorMessage,
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const { error: cancelError, request: cancelJob } = useRequest(
+    useCallback(async () => {
+      setIsOpen(false);
+      await getJobModel(job.type).cancel(job.id);
+    }, [job.id, job.type]),
+    {}
+  );
+  const { error, dismissError: dismissCancelError } = useDismissableError(
+    cancelError
+  );
+
+  return (
+    <>
+      <Tooltip content={title}>
+        {showIconButton ? (
+          <Button
+            aria-label={title}
+            ouiaId="cancel job"
+            onClick={setIsOpen}
+            variant="plain"
+          >
+            <MinusCircleIcon />
+          </Button>
+        ) : (
+          <Button
+            aria-label={title}
+            variant="secondary"
+            ouiaId="cancel job"
+            onClick={setIsOpen}
```

Comment:
We could not find another place that we update a state directly from the event.
```suggestion
             onClick={() => {setIsOpen(open => !open)}}
```